### PR TITLE
Update REVLib new names -> Update REVLib

### PR DESCRIPTION
**Pull request by drorta**

### CODE REVIEW

File: src/main/java/frc/robot/hardware/rev/motors/BrushedSparkMAXMotor.java [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 
-import com.revrobotics.CANSparkLowLevel;
+import com.revrobotics.spark.SparkLowLevel;
 
 public class BrushedSparkMAXMotor extends SparkMaxMotor {
 
 	public BrushedSparkMAXMotor(String logPath, SparkMaxWrapper motor) {
 		super(logPath, motor);
-		if (motor.getMotorType() != CANSparkLowLevel.MotorType.kBrushed) {
+		if (motor.getMotorType() != SparkLowLevel.MotorType.kBrushless) {
```
Comment (hillelv) [inline src/main/java/frc/robot/hardware/rev/motors/BrushedSparkMAXMotor.java:9]: It's supposed to be brushed

File: src/main/java/frc/robot/hardware/rev/motors/SparkMaxMotor.java [hunk 2]
```diff
@@ -65,8 +65,10 @@ public void updateInputs(InputSignal<?>... inputSignals) {
 
 	@Override
 	public void setBrake(boolean brake) {
-		CANSparkBase.IdleMode idleMode = brake ? CANSparkBase.IdleMode.kBrake : CANSparkBase.IdleMode.kCoast;
-		motor.setIdleMode(idleMode);
+		SparkBaseConfig.IdleMode idleMode = brake ? SparkBaseConfig.IdleMode.kBrake : SparkBaseConfig.IdleMode.kCoast;
+
+		//FIXME motor.configure()
+//		motor.setIdleMode(idleMode);
```
Comment (hillelv) [inline src/main/java/frc/robot/hardware/rev/motors/SparkMaxMotor.java:71]: What's the problem with this?

File: src/main/java/frc/robot/hardware/rev/motors/BrushedSparkMAXMotor.java (hunk 1 above)
Comment (drorta) [reply src/main/java/frc/robot/hardware/rev/motors/BrushedSparkMAXMotor.java:9]: oops

File: src/main/java/frc/robot/hardware/rev/motors/SparkMaxMotor.java (hunk 2 above)
Comment (drorta) [reply src/main/java/frc/robot/hardware/rev/motors/SparkMaxMotor.java:71]: This doesnt exist anymore, rev changed the whole way of configuring their devices

More than welcome to look here https://github.com/REVrobotics/REV-Software-Binaries/releases/tag/revlib-2025.0.0-beta-1

תמונה של WhatsApp‏ 2024-12-26 בשעה 19 32 10_f129e4ea

File: src/main/java/frc/robot/hardware/rev/motors/SparkMaxMotor.java [hunk 3]
```diff
@@ -65,8 +65,10 @@ public void updateInputs(InputSignal<?>... inputSignals) {
 
 	@Override
 	public void setBrake(boolean brake) {
-		CANSparkBase.IdleMode idleMode = brake ? CANSparkBase.IdleMode.kBrake : CANSparkBase.IdleMode.kCoast;
-		motor.setIdleMode(idleMode);
+		SparkBaseConfig.IdleMode idleMode = brake ? SparkBaseConfig.IdleMode.kBrake : SparkBaseConfig.IdleMode.kCoast;
+
+		//FIXME motor.configure()
```
Comment (YHerm) [inline src/main/java/frc/robot/hardware/rev/motors/SparkMaxMotor.java:70]: Make the fix me a todo

File: src/main/java/frc/robot/hardware/rev/motors/SparkMaxWrapper.java [hunk 4]
```diff
@@ -1,14 +1,16 @@
...
 import frc.utils.Conversions;
 
-public class SparkMaxWrapper extends CANSparkMax {
+public class SparkMaxWrapper extends SparkMax {
 
 	public SparkMaxWrapper(SparkMaxDeviceID deviceID) {
 		super(deviceID.id(), deviceID.type());
-		super.restoreFactoryDefaults();
+
+		//FIXME super.configure()
```
Comment (YHerm) [inline src/main/java/frc/robot/hardware/rev/motors/SparkMaxWrapper.java:12]: Todo

**Review by YHerm — CHANGES_REQUESTED** [review]

**Review by YHerm — APPROVED** [review]

**Pull request merged**